airbyte-integrations/connectors/source-intercom-asaas/source_intercom/source.py

```diff
@@ -0,0 +1,532 @@
+    """Return list of all admins.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-admins
+    Endpoint: https://api.intercom.io/admins
+    """
+
+    data_fields = ["admins"]
+
+    def path(self, **kwargs) -> str:
+        return "admins"
+
+
+class Companies(IncrementalIntercomStream):
+    """Return list of all companies.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#iterating-over-all-companies
+    Endpoint: https://api.intercom.io/companies/scroll
+    """
+
+    def next_page_token(
+        self, response: requests.Response
+    ) -> Optional[Mapping[str, Any]]:
+        """For reset scroll needs to iterate pages untill the last.
+        Another way need wait 1 min for the scroll to expire to get a new list for companies segments."""
+
+        data = response.json().get("data")
+
+        if data:
+            return {"scroll_param": response.json().get("scroll_param")}
+
+    def path(self, **kwargs) -> str:
+        return "companies/scroll"
+
+
+class CompanySegments(ChildStreamMixin, IncrementalIntercomStream):
+    """Return list of all company segments.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-attached-segments-1
+    Endpoint: https://api.intercom.io/companies/<id>/segments
+    """
+
+    parent_stream_class = Companies
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"/companies/{stream_slice['id']}/segments"
+
+
+class Conversations(IncrementalIntercomStream):
+    """Return list of all conversations.
+
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference/search-for-conversations  # noqa: E501
+    Endpoint: https://api.intercom.io/conversations/search
+    """
+
+    data_fields = ["conversations"]
+    cursor_field = "updated_at"
+
+    def path(self, **kwargs) -> str:
+        """Return the method 'conversations/search'.
+
+        Args:
+            kwargs: kwargs
+
+        Returns:
+            path(str): path
+        """
+        return "conversations/search"
+
+    @property
+    def http_method(self) -> str:
+        """Return method POST.
+
+        Returns:
+            http_method(str): HTTP Method
+        """
+        return "POST"
+
+    def request_body_json(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Optional[Mapping]:
+        """Write the JSON request body to query, sort and paginate results.
+
+        Args:
+            stream_state (Mapping[str, Any]): stream state.
+            stream_slice (Mapping[str, Any], optional):
+                Stream slice. Defaults to None.
+            next_page_token (Mapping[str, Any], optional):
+                Next page token. Defaults to None.
+
+        Returns:
+            Optional[Mapping]: request body json
+        """  # noqa: RST201,RST301
```

Comment:
```suggestion
        """Write the JSON request body to query, sort and paginate results.

        Args:
            stream_state (Mapping[str, Any]): stream state.
            stream_slice (Mapping[str, Any], optional): Stream slice. Defaults to None.
            next_page_token (Mapping[str, Any], optional): Next page token. Defaults to None.

        Returns:
            Optional[Mapping]: request body json
        """
```
Remover os `noqa` e manter na mesma linha deixando o warning de tamanho máximo da linha.